rust-client/examples/challenge_response.rs

```diff
@@ -5,13 +5,16 @@ extern crate veraison_apiclient;
 
 use veraison_apiclient::*;
 
-fn my_evidence_builder(nonce: &[u8], accept: &[String]) -> Result<(Vec<u8>, String), Error> {
+fn my_evidence_builder(nonce: &[u8], accept: &[String], token: Vec<u8>) -> Result<(Vec<u8>, String), Error> {
```

Comment:
@chendave : the evidence builder callback function template is provided by Veraison API client both golang and Rust interface. I do not think, we can modify the template of this call back, like this, as the Execution will break. 

The correct way to implement this is to :
Call another helper function, inside the my_evidence_builder() to construct a real token.
OR
Inline the code to construct a real token and replace it with fake evidence.

Reply:
> the evidence builder callback function template is provided by Veraison API client both golang and Rust interface. 

Oh, I see, here is the interface I think: 
https://github.com/veraison/apiclient/blob/main/verification/evidencebuilder.go#L11

> Call another helper function, inside the my_evidence_builder() to construct a real token.

Due to my limited knowledge on the rust, I tried several ways but no lucky, rust has more restriction than go, pls see: https://gist.github.com/chendave/7d9650c4b6184c0a67539aa84c7829a1

> Inline the code to construct a real token and replace it with fake evidence.

current code flow doesn't allow us to  construct a real token inside the `my_evidence_builder`, the token is a value passed in instead of something we can construct here, I thought about a static variable to make it global, but that seems like is not a good pattern.


I will use this modified patch in my forked repo and have a try, I am not really need to care about the golang interface.